code/modules/projectiles/guns/projectile/pistol.dm

```diff
@@ -8,10 +9,80 @@
 	fire_sound = 'sound/weapons/Gunshot_light.ogg'
 	load_method = MAGAZINE
 
+//For some reason you have to add one of these "if else" statements on the M1911 reskins or the code breaks.
+//If you're not working with two different sprites, just set them to the same icon_state.
+
 /obj/item/weapon/gun/projectile/colt/detective
 	desc = "A Martian recreation of an old Terran pistol. Uses .45 rounds."
 	magazine_type = /obj/item/ammo_magazine/c45m/rubber
 
+/obj/item/weapon/gun/projectile/colt/detective/update_icon()

```

Comment:
Why are there seven separate update_icon()s that appear to be for the exact same type?  Assuming that this works, a much better way to have this work would be to first make all the icon_states with empty mags have '-e' at the end, then you can simply do icon_state = "[unique_reskin]-e" if the gun is empty, with one update_icon(), and if the gun is not empty, then set icon_state = unique_reskin .